Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -582,7 +582,7 @@ UniqueRef<LineContent> LineBuilder::placeInlineAndFloatContent(const InlineItemR
                 // in order to exactly fill the line box. Unless otherwise specified by text-align-last,
                 // the last line before a forced break or the end of the block is start-aligned.
                 auto hasTextAlignJustify = (isLastInlineContent || m_line.runs().last().isLineBreak()) ? rootStyle.textAlignLast() == TextAlignLast::Justify : rootStyle.textAlign() == TextAlignMode::Justify;
-                if (hasTextAlignJustify) {
+                if (hasTextAlignJustify && rootStyle.textJustify() != TextJustify::None) {
```

Comment:
according to the spec `text-justify: none` sets the justification opportunities to 0 (https://drafts.csswg.org/css-text/#text-justify-property). does this mean ruby align is impacted too as it aligns the base content using the number of justification opportunities? (https://drafts.csswg.org/css-ruby/#ruby-align-property)